lambdaclass/mirra_backend
# [GH-580] Fake Item and Traps

## Motivation

Add the new fake item to the game and also add the new traps feature. 

Closes #580
After merging this, lets merge https://github.com/lambdaclass/curse_of_mirra/pull/1772 into curse of mirra's front

## Summary of changes

- Implemented items mechanics
- Added new item `Fake Item`
- Added `Arena.Game.Item` and `Arena.Game.Trap` modules
- Added `Trap` and `TrapStatus` message to `messages.proto`

## Demo

https://github.com/lambdaclass/mirra_backend/assets/67444519/7dbfeded-5122-44c1-b423-a3c1b89348ae

## How to test it?
You'll see that now an item could've this fields. Traps are entities too, so they pretty much will have the same parameters as any other entity. In the code you'll which fields are part of the additional info

```json
  {
    "name": "fake_item",
    "radius": 200.0,
    "effects": [],
    "mechanics": {
      "spawn_bomb": {
        "name": "bomb",
        "radius": 200.0,
        "activation_delay_ms": 3000,
        "vertices": [],
        "mechanics": {
          "circle_hit": {
            "damage": 64,
            "range": 380.0,
            "offset": 400
          }
        }
      }
    }
```

They're pretty much self explanatory, but maybe we should add a field that addresses animation time.

- Get a Fake Item
- Use it nearby an enemy and it will leave a bomb
- Wait until explodes
- Boom!

## Checklist
